cpp/src/plasma/common.cc

```diff
@@ -68,12 +68,57 @@ std::string UniqueID::hex() const {
   return result;
 }
 
-size_t UniqueID::hash() const {
-  size_t result;
-  std::memcpy(&result, id_, sizeof(size_t));
-  return result;
+// This code is from https://sites.google.com/site/murmurhash/
+// and is public domain.
+uint64_t MurmurHash64A(const void* key, int len, unsigned int seed) {
```

Comment:
At some point it may make sense to move this (and other hash functions) to `src/arrow/util`

Reply:
Yes, this is a common method, but I don't know whether there is other place needing this hash method. Besides, this change is a replacement and is small, we may extract all the common methods in plasma to src/arrow/util later.

Reply:
I created a follow up JIRA for this: https://issues.apache.org/jira/browse/ARROW-2803